notification/events.go

```diff
@@ -324,6 +321,67 @@ func checkInhibition(ctx context.Context, notif NotificationWithSpec, resource t
 			continue
 		}
 
+		inhibitionWindow := ctx.Properties().Duration("notification.inhibition.window", DefaultInhibitionWindow)
+		if inhibition.Interval != nil {
+			w, err := duration.ParseDuration(*inhibition.Interval)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse inhibition interval: %w", err)
+			}
+
+			inhibitionWindow = time.Duration(w)
+		}
+
+		if len(inhibition.GroupBy) > 0 {
+			var sendHistories []struct {
+				ID         uuid.UUID
+				ResourceID uuid.UUID
+			}
+
+			var resourceTable string
+			if strings.HasPrefix(payload.EventName, "config") {
+				resourceTable = "config_items"
+			} else if strings.HasPrefix(payload.EventName, "component") {
+				resourceTable = "components"
+			} else if strings.HasPrefix(payload.EventName, "check") {
+				resourceTable = "checks"
+			} else {
+				return nil, fmt.Errorf("unsupported event type: %s", payload.EventName)
+			}
+
+			query := fmt.Sprintf(`
+				SELECT notification_send_history.id, resource_id
+				FROM notification_send_history
+				INNER JOIN %s on resource_id = %s.id and %s.type = ?
+				WHERE 
+				notification_send_history.notification_id = ?
+				AND notification_send_history.status = ?
+				AND notification_send_history.created_at >= NOW() - INTERVAL '%f MINUTES'`, resourceTable, resourceTable, resourceTable, inhibitionWindow.Minutes())
+			if err := ctx.DB().Raw(query, inhibition.From, notif.ID, models.NotificationStatusSent).Scan(&sendHistories).Error; err != nil {
+				return nil, fmt.Errorf("failed to get notification send history: %w", err)
+			}
+
+			if len(sendHistories) > 0 {
+				requiredHash, err := calculateGroupByHash(ctx, inhibition.GroupBy, payload.ID.String(), payload.EventName)
+				if err != nil {
+					return nil, fmt.Errorf("failed to calculate group by hash: %w", err)
+				}
+
+				for _, sendHistory := range sendHistories {
+					hash, err := calculateGroupByHash(ctx, inhibition.GroupBy, sendHistory.ResourceID.String(), payload.EventName)
+					if err != nil {
+						return nil, fmt.Errorf("failed to calculate group by hash: %w", err)
+					}
+
+					if hash == requiredHash {
+						return lo.ToPtr(sendHistory.ID), nil
+					}
+				}
+			}
```

Comment:
We either

1. precalculate the hash for all inhibition rules and save it in the send history (like we do with group_by_hash). The problem with that is
- Doesn't work if inhibition rule changes
- we'd have to store 1 groupByHash per inhibition rule (maybe in a array column). But, w'd need to id which inhibition rule a groupByHash in the list is for. So the inhibition rule would need a name also?

2. Calculate the hash on-demand. Flexible with changing inhibition rules but inefficient as the number of inhibition rules/window grows